test_invalid_utf8 fails very rarely

In #66, a commit that changed a docstring caused test_invalid_utf8 to fail.

To debug this, I wrote a loop that runs pytest until it fails:

```
while python3 -m pytest -v tests/test_mantatail.py::test_invalid_utf8; do :; done
```

It can take many attempts to get the tests to fail. When it eventually does fail, the error looks e.g. like this:

```
======================================== test session starts =========================================
platform linux -- Python 3.9.2, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /home/akuli/MantaTail/env/bin/python3
cachedir: .pytest_cache
rootdir: /home/akuli/MantaTail, configfile: pytest.ini
collected 1 item                                                                                     

tests/test_mantatail.py::test_invalid_utf8 Mantatail running (127.0.0.1:6667)
FAILED

============================================== FAILURES ==============================================
_________________________________________ test_invalid_utf8 __________________________________________

user_alice = <socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 36850), raddr=('127.0.0.1', 6667)>
user_bob = <socket.socket fd=9, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 36852), raddr=('127.0.0.1', 6667)>

    def test_invalid_utf8(user_alice, user_bob):
        user_alice.sendall(b"JOIN #foo\r\n")
        time.sleep(0.1)
        user_bob.sendall(b"JOIN #foo\r\n")
    
        while receive_line(user_alice) != b":Bob!BobUsr@127.0.0.1 JOIN #foo\r\n":
            pass
        while receive_line(user_bob) != b":mantatail 366 Bob #foo :End of /NAMES list.\r\n":
            pass
    
        random_message = os.urandom(100).replace(b"\n", b"")
        user_alice.sendall(b"PRIVMSG #foo :" + random_message + b"\r\n")
>       assert receive_line(user_bob) == b":Alice!AliceUsr@127.0.0.1 PRIVMSG #foo :" + random_message + b"\r\n"
E       AssertionError: assert b':Alice!Alic...cMN-z\x81\r\n' == b':Alice!Alic...cMN-z\x81\r\n'
E         At index 40 diff: b'\xb3' != b':'
E         Full diff:
E           (
E         -  b':Alice!AliceUsr@127.0.0.1 PRIVMSG #foo ::\xb3t\xdb\xb7\xbb\xd7\xec'
E         ?                                           -
E         +  b':Alice!AliceUsr@127.0.0.1 PRIVMSG #foo :\xb3t\xdb\xb7\xbb\xd7\xec\xd1'
E         ?                                                                     ++++...
E         
E         ...Full output truncated (17 lines hidden), use '-vv' to show

tests/test_mantatail.py:578: AssertionError
====================================== short test summary info =======================================
FAILED tests/test_mantatail.py::test_invalid_utf8 - AssertionError: assert b':Alice!Alic...cMN-z\x8...
========================================= 1 failed in 0.16s ==========================================
```

Note that I have added `-v` to my command. Without `-v`, pytest doesn't show much of the bytes being compared.

The error message is a bit hard to read, but it is saying that the text expected to receive `PRIVMSG #foo ::\xb3t\xdb` while it actually received `PRIVMSG #foo :\xb3t\xdb`.

So to me it seems like you can't send a message starting with `:` with Mantatail: if you do, instead of sending `PRIVMSG #foo ::hello` (first `:` means the rest is the message, second `:` is a part of the message `:hello`), Mantatail will send `PRIVMSG #foo :hello` (the only `:` means the rest is message, and the message is now `hello`).

--- a/commands.py
+++ b/commands.py
@@ -359,7 +359,7 @@ def parse_received_args(msg: str) -> Tuple[str, List[str]]:
     for num, arg in enumerate(split_msg):
         if arg.startswith(":"):
             parsed_msg = split_msg[:num]
-            parsed_msg.append(" ".join(split_msg[num:]).lstrip(":"))
+            parsed_msg.append(" ".join(split_msg[num:])[1:])
             command = parsed_msg[0]
             return command, parsed_msg[1:]
 

--- a/tests/test_mantatail.py
+++ b/tests/test_mantatail.py
@@ -587,3 +587,19 @@ def test_invalid_utf8(user_alice, user_bob):
     random_message = os.urandom(100).replace(b"\n", b"")
     user_alice.sendall(b"PRIVMSG #foo :" + random_message + b"\r\n")
     assert receive_line(user_bob) == b":Alice!AliceUsr@127.0.0.1 PRIVMSG #foo :" + random_message + b"\r\n"
+
+
+def test_message_starting_with_colon(user_alice, user_bob):
+    user_alice.sendall(b"JOIN #foo\r\n")
+    time.sleep(0.1)
+    user_bob.sendall(b"JOIN #foo\r\n")
+
+    while receive_line(user_alice) != b":Bob!BobUsr@127.0.0.1 JOIN #foo\r\n":
+        pass
+    while receive_line(user_bob) != b":mantatail 366 Bob #foo :End of /NAMES list.\r\n":
+        pass
+
+    # Alice sends ":O lolwat" to Bob.
+    # It is prefixed with a second ":" because of how IRC works.
+    user_alice.sendall(b"PRIVMSG #foo ::O lolwat\r\n")
+    assert receive_line(user_bob) == b":Alice!AliceUsr@127.0.0.1 PRIVMSG #foo ::O lolwat\r\n"

--- a/tests/test_mantatail.py
+++ b/tests/test_mantatail.py
@@ -603,3 +603,7 @@ def test_message_starting_with_colon(user_alice, user_bob):
     # It is prefixed with a second ":" because of how IRC works.
     user_alice.sendall(b"PRIVMSG #foo ::O lolwat\r\n")
     assert receive_line(user_bob) == b":Alice!AliceUsr@127.0.0.1 PRIVMSG #foo ::O lolwat\r\n"
+
+    # Alice sends ":O"
+    user_alice.sendall(b"PRIVMSG #foo ::O\r\n")
+    assert receive_line(user_bob) == b":Alice!AliceUsr@127.0.0.1 PRIVMSG #foo ::O\r\n"

